lib/include/openamp/rpmsg_virtio.h

```diff
@@ -29,16 +29,32 @@ extern "C" {
 /* The feature bitmap for virtio rpmsg */
 #define VIRTIO_RPMSG_F_NS 0 /* RP supports name service notifications */
 
+struct rpmsg_virtio_shm_pool;
+/**
+ * struct rpmsg_virtio_shm_pool - shared memory pool used for rpmsg buffers
+ * @get_buffer: function to get buffer from the pool
+ * @base: base address of the memory pool
+ * @avail: available memory size
+ * @size: total pool size
+ */
+struct rpmsg_virtio_shm_pool {
+	void *(*get_buffer)(struct rpmsg_virtio_shm_pool *shpool, size_t size);
+	void *base;
+	size_t avail;
+	size_t size;
+};
+
+/* Default rpmsg virtio shared memory pool */
+extern struct rpmsg_virtio_shm_pool rpmsg_virtio_default_shm_pool;
```

Comment:
This means that you define a common pool for all communications. So if  you have to communicate with 2 separate processors , this impose that in your ssytem  the shared memory is shared between all processors...
=> Need to define this pool in rpmsg_virtio_device struct not as global variable. Application should ensure the coherence if need to share the same memory for several rpmsg IPC links.

Reply:
good point. cannnot have global variable as they are not just operations.